File: 13-python-integration/5-permutations/benchmark.py
#!python3

import itertools
import time

def benchmark(N:int)->int:
	count=0
	print ("Permutations of 1..{}:".format(N), flush=True)
	start = time.clock()
	for p in itertools.permutations(range(1,N)):
		count += 1
	end = time.clock()
	duration_in_seconds = end-start
	print("  {} permutations calculated in {} seconds".format(count, duration_in_seconds), flush=True)


for N in range(5,13+1):
	benchmark(N)

File: 13-python-integration/5-permutations/benchmark.cpp
/**
 * Benchmark for computing all permutations
 */

#include "output_containers.hpp"
#include <algorithm>
#include <numeric>
using namespace std;

void benchmark(int N) {
	vector<int> v(N);
	auto begin = time(NULL);
	iota(v.begin(), v.end(), 1); // fill the vector with 1,...,N
	int count=0;
	cout << "Permutations of 1.." << N << ":" << endl;
	do {
		++count;
	} while ( next_permutation(v.begin(),v.end()) );
	auto end = time(NULL);
	auto duration = end-begin;
	cout << "  " << count << " permutations calculated in " << duration << " seconds" << endl;
}

int main() {
	for (int i=5;i<=13; ++i)
		benchmark(i);
	return 0;
}
